test/test_optim.py

```diff
@@ -495,10 +495,23 @@ def test_multi_tensor_optimizers(self):
 
                     optimizer.step()
 
+                state.append(optimizer.state)
                 res.append(model.parameters())
 
-            for p1, p2 in zip(res[0], res[1]):
-                self.assertEqual(p1, p2, atol=5e-5, rtol=0)
+            st_state = state[0]
+            mt_state = state[1]
+            for st_p, mt_p in zip(res[0], res[1]):
+                self.assertEqual(st_p, mt_p, atol=5e-5, rtol=0)
+
+                # check that optimizer states are the same
+                st_p_state = st_state[st_p]
+                mt_p_state = mt_state[mt_p]
+
+                for k in st_p_state:
+                    # FIXME: remove this when https://github.com/pytorch/pytorch/issues/78807 is resolved
```

Comment:
@albanD this is where it's failing for Adam, but I'm currently skipping it

Reply:
Ho read this too quickly.
And why are the params not affected? Is it just that the number of iteration is too small for this to make a visible difference on the params?

Reply:
Yes, it seems so, I'll try increasing the num iterations and see whether this fails

Reply:
Increasing num_iterations to 300 made some parameter comparisons fail for Adam and AdamW, I am also starting to see errors for NAdam. 

Additionally, comparisons for `exp_avg_sq` and `exp_avg` are starting to fail for these three optimizers. This is odd to me because iiuc, these two buffers are only updated [here](https://github.com/pytorch/pytorch/blob/master/torch/optim/adam.py#L263-L264) using equations like `exp_avg = beta_1 * exp_avg + (1-beta_1) * grad`, so this seems more like some CUDA tolerance errors(?)

I'm currently printing these mismatches in CI to illustrate this and will update this comment with the full list of assertion failures after CI runs

Reply:
Yeah it's probably accuracy errors accumulated over a lot of steps.